app/controllers/settings/profiles_controller.rb

```diff
@@ -23,7 +23,7 @@ def update
   private
 
   def account_params
-    params.require(:account).permit(:display_name, :note, :avatar, :header, :locked, :bot, :discoverable, fields_attributes: [:name, :value])
+    params.require(:account).permit(:display_name, :note, :avatar, :header, :locked, :bot, :discoverable, :subaction, fields_attributes: [:name, :value])
```

Comment:
I think you kind of overcomplicated it. I think having separate routes for clearing avatar/header, and just POSTing to those routes through the `link_to` helper, and reloading the whole page would be fine.